File: 2305-append-k-integers-with-minimal-sum/append-k-integers-with-minimal-sum.java
class Solution {
    public long minimalKSum(int[] nums, int k) {
        Arrays.sort(nums);
    long sum = 0;
    long current = 1;
    
    for (int i = 0; i < nums.length && k > 0; i++) {
        // Skip duplicates
        if (i > 0 && nums[i] == nums[i - 1]) continue;
        
        // Count of numbers we can take before nums[i]
        long count = Math.min(k, Math.max(0, (long)nums[i] - current));
        
        // Sum of arithmetic series: (first + last) * count / 2
        sum += (current + current + count - 1) * count / 2;
        k -= count;
        
        // Next start number after nums[i]
        current = (long)nums[i] + 1;
    }
    
    // If k still remains, continue after last number
    if (k > 0) {
        sum += (current + current + k - 1) * k / 2;
    }
    
    return sum;
    }
}

File: 2305-append-k-integers-with-minimal-sum/append-k-integers-with-minimal-sum.cpp
class Solution {
public:
    long long minimalKSum(vector<int>& nums, int k) {
        int n=nums.size();
        long long total=0;
        sort(nums.begin(),nums.end());
        int prev=0;

        for(int i=0; i<n; i++){
            int interval=min(nums[i]-prev-1,k);
            int curr=prev+interval;
            if(interval>0){
                long long prevSum=1ll*prev*(prev+1)/2;
                long long currSum=1ll*curr*(curr+1)/2;
                long long sum=currSum-prevSum;

                total+=sum;
                k-=interval;
            }
            prev=nums[i];
        }

        if(k){
            int prev=nums.back();
            int curr=prev+k;
            long long prevSum=1ll*prev*(prev+1)/2;
            long long currSum=1ll*curr*(curr+1)/2;
            long long sum=currSum-prevSum;

            total+=sum;
        }

        return total;
    }
};